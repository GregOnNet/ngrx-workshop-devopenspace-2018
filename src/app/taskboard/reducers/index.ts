import { ActionReducerMap } from '@ngrx/store';
import * as fromRoot from '../../reducers';
import * as fromTasks from './taskboard.reducer';

export interface TaskboardState {
  tasks: fromTasks.Slice;
}

export interface State extends fromRoot.State {
  taskboard: TaskboardState;
}

export const reducers: ActionReducerMap<TaskboardState> = {
  tasks: fromTasks.reducer
};
