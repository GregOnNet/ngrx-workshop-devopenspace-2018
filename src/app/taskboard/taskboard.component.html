<app-quick-add (create)="buildTask($event)"></app-quick-add>
