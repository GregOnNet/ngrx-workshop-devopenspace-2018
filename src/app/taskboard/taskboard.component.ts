import { Component, OnInit } from '@angular/core';
import { Store } from '@ngrx/store';
import { newGuid } from 'ts-guid';
import { Task, TaskDraft } from './models';

@Component({
  selector: 'app-taskboard',
  templateUrl: './taskboard.component.html',
  styleUrls: ['./taskboard.component.css']
})
export class TaskboardComponent implements OnInit {
  constructor(private _store: Store<any>) {}

  ngOnInit() {}

  buildTask(draft: TaskDraft) {
    this._store.dispatch(
      new CreateTask({
        guid: newGuid(),
        isInProgress: false,
        isDone: false,
        ...draft
      })
    );
  }
}

export class CreateTask {
  readonly type = 'Create';

  constructor(public payload: Task) {}
}
