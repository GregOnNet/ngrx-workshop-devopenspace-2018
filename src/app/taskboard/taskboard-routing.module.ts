import { NgModule } from '@angular/core';
import { Route, RouterModule } from '@angular/router';
import { TaskDetailsComponent } from './containers/task-details/task-details.component';
import { TasksDashboardComponent } from './containers/tasks-dashboard/tasks-dashboard.component';
import { TaskboardComponent } from './taskboard.component';

const routes: Route[] = [
  {
    path: '',
    component: TaskboardComponent,
    children: [
      { path: '', component: TasksDashboardComponent },
      { path: ':guid', component: TaskDetailsComponent }
    ]
  }
];

@NgModule({ imports: [RouterModule.forChild(routes)], exports: [RouterModule] })
export class TaskboardRoutingModule {}
