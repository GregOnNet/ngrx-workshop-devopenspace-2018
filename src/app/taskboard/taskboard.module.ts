import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule
} from '@angular/material';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';
import { TaskboardEffects } from './effects/taskboard.effects';
import { QuickAddComponent } from './quick-add/quick-add.component';
import * as fromTaskboard from './reducers';
import { TaskCardComponent } from './task-card/task-card.component';
import { TaskboardComponent } from './taskboard.component';

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    MatCardModule,
    MatInputModule,
    MatFormFieldModule,
    MatButtonModule,
    StoreModule.forFeature('taskboard', fromTaskboard.reducers),
    EffectsModule.forFeature([TaskboardEffects])
  ],
  declarations: [TaskboardComponent, QuickAddComponent, TaskCardComponent]
})
export class TaskboardModule {}
