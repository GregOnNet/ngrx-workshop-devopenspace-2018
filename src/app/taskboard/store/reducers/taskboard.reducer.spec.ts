import { Task } from '../../models';
import {
  CreateTaskSucceeded,
  TaskboardActionTypes
} from '../actions/taskboard.actions';
import { reducer, Slice } from './taskboard.reducer';

describe('Taskboard Actions', () => {
  describe(`${TaskboardActionTypes.Create}`, () => {
    describe('When action is raised', () => {
      it('should create a task', () => {
        const task = { guid: '123' } as Task;
        const initialState: Slice = { entities: {} };
        const action = new CreateTaskSucceeded(task);

        const state = reducer(initialState, action);

        expect(state).toMatchSnapshot();
      });
    });

    describe('When no guid is provided', () => {
      it('should not create a task', () => {
        const taskWithoutGuid = { } as Task;
        const initialState: Slice = { entities: {} };
        const action = new CreateTaskSucceeded(taskWithoutGuid);

        const state = reducer(initialState, action);

        expect(state).toMatchSnapshot();
      });
    });
  });
});
