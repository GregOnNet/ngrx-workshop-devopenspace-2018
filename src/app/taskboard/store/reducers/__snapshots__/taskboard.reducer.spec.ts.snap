// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taskboard Actions [Taskboard] Create new Task succeeded When action is raised should create a task 1`] = `
Object {
  "entities": Object {
    "123": Object {
      "guid": "123",
    },
  },
}
`;

exports[`Taskboard Actions [Taskboard] Create new Task succeeded When no guid is provided should not create a task 1`] = `
Object {
  "entities": Object {},
}
`;
